cohortextractor/presto_utils.py

```diff
@@ -1,19 +1,47 @@
+import os
 import readline  # noqa -- importing this adds readline behaviour to input()
 import time
 from urllib.parse import urlparse, unquote
 
 import prestodb
 import requests
+from requests_pkcs12 import Pkcs12Adapter
 from tabulate import tabulate
 
 
 def presto_connection_from_url(url):
-    return ConnectionProxy(
-        prestodb.dbapi.connect(**presto_connection_params_from_url(url))
+    """Return a connection to Presto instance at given URL."""
+
+    conn_params = presto_connection_params_from_url(url)
+    conn = prestodb.dbapi.connect(**conn_params)
+    if "PFX_PATH" in os.environ:
+        adapt_connection(conn, conn_params)
+
+    return ConnectionProxy(conn)
+
+
+def adapt_connection(conn, conn_params):
+    """Adapt connection to use passphrase-protected PKCS#12 certificate.
+
+    See instructions in opensafely/accessing-emis-data/ for getting a certificate.
+    """
+
+    with open(os.environ["PFX_PASSWORD_PATH"], "rb") as f:
+        pkcs12_password = f.read().strip()
+
+    session = requests.Session()
+    mount_prefix = "{http_scheme}://{host}:{port}".format(**conn_params)
+    mount_adaptor = Pkcs12Adapter(
+        pkcs12_filename=os.environ["PFX_PATH"], pkcs12_password=pkcs12_password,
     )
+    session.mount(mount_prefix, mount_adaptor)
+    session.verify = False
+    conn._http_session = session
```

Comment:
Ugh, I looked at the library code for this, unpleasant. Nothing we can do though